### ISSUE TITLE
Map.get_by_coord() fails for non float data type

### ISSUE THREAD

**Issue by registerrier** [gammapy/gammapy#2074]
`Map.get_by_coord()` returns an error when `dtype='int'`, because it sets the result to `np.nan` which is wrongly interpreted when the `dtype` is not `float`:

```
res=WcsNDMap.create(width=2,dtype='int') 
coords={'lon':[1,3],'lat':[1.,0.]}  
res.get_by_coord(coords) 
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-55c0d99d4986> in <module>
----> 1 res.get_by_coord(coords)

~/Code/gammapy-dev/gammapy/gammapy/maps/base.py in get_by_coord(self, coords)
    713         idx = self.geom.coord_to_idx(coords)
    714         vals[msk] = self.get_by_idx(idx)
--> 715         vals[~msk] = np.nan
    716         return vals
    717 

ValueError: cannot convert float NaN to integer
```

Even worse if the `dtype = 'bool'` (e.g. map mask):
```
res=WcsNDMap.create(width=2,dtype='bool') 
coords={'lon':[1,3],'lat':[1.,0.]}  
res.get_by_coord(coords) 
Out[12]: array([ True,  True])
```
where the expected result should be `False` because both points are outside the map.

**Comment by adonath** [comment]
Related #1440.

**Comment by adonath** [comment]
Resolved by #2078.
